src/reanimated2/PropAdapters.ts

```diff
@@ -1,18 +1,25 @@
 import { addWhitelistedNativeProps } from '../ConfigHelper';
 import type { AdapterWorkletFunction } from './commonTypes';
+import type { PropsAdapterFunction } from './helperTypes';
 
-export function createAnimatedPropAdapter(
+// TODO TYPESCRIPT This is a temporary type to get rid of .d.ts file.
+export type createAnimatedPropAdapterType = (
+  adapter: PropsAdapterFunction,
+  nativeProps?: string[]
+) => PropsAdapterFunction;
+
+export const createAnimatedPropAdapter = ((
   adapter: AdapterWorkletFunction,
   nativeProps?: string[]
-): AdapterWorkletFunction {
+): AdapterWorkletFunction => {
   const nativePropsToAdd: { [key: string]: boolean } = {};
   // eslint-disable-next-line no-unused-expressions
   nativeProps?.forEach((prop) => {
     nativePropsToAdd[prop] = true;
   });
   addWhitelistedNativeProps(nativePropsToAdd);
   return adapter;
-}
+}) as createAnimatedPropAdapterType;
```

Comment:
Why do we need to cast the whole function instead of just casting the return value?

Reply:
If it is not a 'hot fix' I think we should avoid type casting